src/tracing.rs

```diff
@@ -1,55 +1,40 @@
-pub fn setup(log_level: &str) -> anyhow::Result<()> {
-    // Use `console` and/or `tracy` features to enable both `tracing-subscriber` and the layer(s)
-    #[cfg(feature = "tracing-subscriber")]
-    {
-        use tracing_subscriber::prelude::*;
-
-        let reg = tracing_subscriber::registry();
-
-        // Use `console` feature to enable both `tracing-subscriber` and `console-subscriber`
-        #[cfg(feature = "console-subscriber")]
-        let reg = reg.with(console_subscriber::spawn());
-
-        // Note, that `console-subscriber` requires manually enabling
-        // `--cfg tokio_unstable` rust flags during compilation!
-        //
-        // Otherwise `console_subscriber::spawn` call panics!
-        //
-        // See https://docs.rs/tokio/latest/tokio/#unstable-features
-        #[cfg(all(feature = "console-subscriber", not(tokio_unstable)))]
-        eprintln!(
-            "`console-subscriber` requires manually enabling \
-             `--cfg tokio_unstable` rust flags during compilation!"
-        );
-
-        // Use `tracy` feature to enable both `tracing-subscriber` and `tracing-tracy`
-        #[cfg(feature = "tracing-tracy")]
-        let reg = reg.with(tracing_tracy::TracyLayer::new().with_filter(
-            tracing_subscriber::filter::filter_fn(|metadata| metadata.is_span()),
-        ));
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::{filter, fmt};
 
-        #[cfg(all(feature = "tracing-log-always", feature = "tracing-logger"))]
-        eprintln!(
-            "Both `tracing-log-always` and `tracing-logger` features are enabled at the same time. \
-             This will cause some logs to be printed twice!"
-        );
-
-        #[cfg(feature = "tracing-logger")]
-        let reg = reg.with(
-            tracing_subscriber::fmt::layer()
-                .with_ansi(true)
-                .with_span_events(tracing_subscriber::fmt::format::FmtSpan::NEW)
-                .with_filter(
-                    tracing_subscriber::filter::EnvFilter::builder().parse_lossy(log_level),
-                ),
-        );
-
-        #[cfg(feature = "tracing-logger")]
-        tracing_log::LogTracer::init()?;
-
-        tracing::subscriber::set_global_default(reg)?;
-    }
+pub fn setup(log_level: &str) -> anyhow::Result<()> {
+    tracing_log::LogTracer::init()?;
+
+    let reg = tracing_subscriber::registry().with(
+        fmt::layer()
+            .with_ansi(true)
+            .with_span_events(fmt::format::FmtSpan::NEW)
+            .with_filter(filter::EnvFilter::builder().parse_lossy(log_level)),
+    );
+
+    // Use `console` or `console-subscriber` feature to enable `console-subscriber`
+    //
+    // Note, that `console-subscriber` requires manually enabling
+    // `--cfg tokio_unstable` rust flags during compilation!
+    //
+    // Otherwise `console_subscriber::spawn` call panics!
+    //
+    // See https://docs.rs/tokio/latest/tokio/#unstable-features
```

Comment:
Could you elaborate a bit on what this means. Is this only relevant for interacting with Tokio Console?

Reply:
> Is this only relevant for interacting with Tokio Console?

TL;DR: Yes.

There's a `tracing` feature in Tokio, but it's considered _experimental_, so if you want to enable it, you have to also explicitly set `RUSTFLAGS='--cfg tokio_unstable'`. And, this feature/flag combo simply enables `tracing` in Tokio. Like, tracing-spans/additional logs/etc.

It's _required_ for Tokio Console, but it _might_ also be useful for general Tokio debugging sometime. If you only want Tokio `tracing`, but without Tokio Console, you can enable `tokio-tracing` feature of the `qdrant` crate, but you will still have to set `RUSTFLAGS='--cfg tokio_unstable'`.